src/Libraries/Microsoft.Extensions.AI/ChatCompletion/ImageGeneratingChatClient.cs

```diff
@@ -0,0 +1,517 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>A delegating chat client that enables image generation capabilities by converting <see cref="HostedImageGenerationTool"/> instances to function tools.</summary>
+/// <remarks>
+/// <para>
+/// The provided implementation of <see cref="IChatClient"/> is thread-safe for concurrent use so long as the
+/// <see cref="IImageGenerator"/> employed is also thread-safe for concurrent use.
+/// </para>
+/// <para>
+/// This client automatically detects <see cref="HostedImageGenerationTool"/> instances in the <see cref="ChatOptions.Tools"/> collection
+/// and replaces them with equivalent function tools that the chat client can invoke to perform image generation and editing operations.
+/// </para>
+/// </remarks>
+[Experimental("MEAI001")]
+public sealed class ImageGeneratingChatClient : DelegatingChatClient
+{
+    /// <summary>
+    /// Specifies how image and other data content is handled when passing data to an inner client.
+    /// </summary>
+    /// <remarks>
+    /// Use this enumeration to control whether images in the data content are passed as-is, replaced
+    /// with unique identifiers, or only generated images are replaced. This setting affects how downstream clients
+    /// receive and process image data.
+    /// Reducing what's passed downstream can help manage the context window.
+    /// </remarks>
+    public enum DataContentHandling
```

Comment:
The purpose of this is to control which images get removed from messages (to be treated only candidates for edit) vs which are sent across (to be candidates both for edit and  vision / model understanding).

The reason for doing this is to avoid blowing out the cotnext window with very large images that are never meant to be consumed by the model.